pulp_service/pulp_service/app/viewsets.py

```diff
@@ -146,21 +147,29 @@ def view_name(cls):
         return "admintasks"
 
 
-class VulnerabilityReport(ViewSet):
+class VulnerabilityReport(NamedModelViewSet, ListModelMixin, RetrieveModelMixin):
 
-    def list(self, request):
-        queryset = VulnReport.objects.all()
-        serializer = VulnerabilityReportSerializer(queryset, many=True)
-        return Response(serializer.data)
+    endpoint_name = "vuln_report"
+    queryset = VulnReport.objects.all()
+    serializer_class = VulnerabilityReportSerializer
 
-    def get(self, request, uuid):
-        queryset = VulnReport.objects.get(id=uuid)
-        serializer = VulnerabilityReportSerializer(queryset, many=False)
-        return Response(serializer.data)
-
-    def post(self, request):
-        serialized_data = ContentScanSerializer(data=request.data)
+    def create(self, request):
+        serialized_data = ContentScanSerializer(data=request.data, context={'request': request})
         serialized_data.is_valid(raise_exception=True)
-        repo_version_pk = serialized_data.data["repo_version"]
-        task = dispatch(check_content, kwargs={"repo_version_pk": repo_version_pk})
+
+        shared_resources = None
+        """Dispatch a task to scan the Content Units from a Repository"""
+        if repo_version_hyperlink := serialized_data.data.get("repo_version", None):
+            repository_version = repo_version_hyperlink.obj
```

Comment:
After you call `is_valid` you typically get the validated data using `serializer.validated_data.get("repo_version")` and this returned value will be a RepositoryVersion (at least it should be)